src/integrationTest/java/uk/gov/hmcts/reform/ccd/documentam/controller/CaseDocumentAmControllerIT.java

```diff
@@ -0,0 +1,326 @@
+import static uk.gov.hmcts.reform.ccd.documentam.fixtures.WiremockFixtures.stubDocumentUrl;
+import static uk.gov.hmcts.reform.ccd.documentam.fixtures.WiremockFixtures.stubGetDocumentMetaData;
+import static uk.gov.hmcts.reform.ccd.documentam.fixtures.WiremockFixtures.stubPatchDocument;
+import static uk.gov.hmcts.reform.ccd.documentam.fixtures.WiremockFixtures.stubPatchDocumentMetaData;
+import static uk.gov.hmcts.reform.ccd.documentam.fixtures.WiremockFixtures.stubUploadDocument;
+
+@RunWith(SpringRunner.class)
+
+public class CaseDocumentAmControllerIT extends BaseTest {
+
+    @Value("${idam.s2s-auth.totp_secret}")
+    protected String salt;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private static final String MAIN_URL = "/cases/documents";
+    private static final String ATTACH_TO_CASE_URL = "/attachToCase";
+    private static final String SERVICE_NAME_XUI_WEBAPP = "xui_webapp";
+    private static final String SERVICE_NAME_CCD_DATA = "ccd_data";
+
+    private static final String CASE_ID_KEY = "caseId";
+    private static final String CLASSIFICATION_KEY = "classification";
+    private static final String CASE_TYPE_ID_KEY = "caseTypeId";
+    private static final String JURISDICTION_ID_KEY = "jurisdictionId";
+
+    private static final String CLASSIFICATION = "PUBLIC";
+    private static final String CASE_TYPE_ID = "BEFTA_CASETYPE_2";
+    private static final String JURISDICTION_ID = "BEFTA_JURISDICTION_2";
+
+    private static final String CASE_ID = "1584722156538291";
+    private static final String USER_ID = "d5566a63-f87c-4658-a4d6-213d949f8415";
+
+    @Test
+    void shouldSuccessfullyUploadDocument() throws Exception {
+
+        stubDocumentUrl();
+        stubUploadDocument();
+        stubDocumentManagementUploadDocument();
+
+        MockMultipartFile firstFile =
+            new MockMultipartFile("files", "filename.txt",
+                                  "text/plain", "some xml".getBytes());
+        MockMultipartFile secondFile =
+            new MockMultipartFile("data", "other-file-name.data",
+                                  "text/plain", "some other type".getBytes());
+        MockMultipartFile jsonFile =
+            new MockMultipartFile("json", "",
+                                  MediaType.APPLICATION_JSON_VALUE, "{\"json\": \"someValue\"}".getBytes());
+
+        mockMvc.perform(MockMvcRequestBuilders.multipart(MAIN_URL)
+                            .file(firstFile)
+                            .file(secondFile)
+                            .file(jsonFile)
+                            .headers(createHttpHeaders(SERVICE_NAME_XUI_WEBAPP))
+                            .param(CLASSIFICATION_KEY, CLASSIFICATION)
+                            .param(CASE_TYPE_ID_KEY, CASE_TYPE_ID)
+                            .param(JURISDICTION_ID_KEY, JURISDICTION_ID)
+                            .contentType(MediaType.MULTIPART_FORM_DATA_VALUE))
+            .andExpect(status().isOk())
```

Comment:
Can you assert response content as well ? https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1456373817&__ncforminfo=FxYVXDsMWt_l-Nsq7GKZ23KbWuWfyp45H7K83x4Uq4rnyEoN-SWdvP_8LVJNx2AGDom5mTLlA94h-xAUpU8twJYMBrAma_sE